datatable/java/datatable/src/main/java/io/cucumber/datatable/SimpleDefaultDataTableEntryTransformer.java

```diff
@@ -0,0 +1,71 @@
+package io.cucumber.datatable;
+
+import java.lang.reflect.Field;
+import java.util.Map;
+
+/**
+ * {@link DefaultDataTableEntryTransformer} based on reflection.
+ * <p>
+ * For example:
+ *
+ * <pre>
+ *  | firstName   | last name | age |
+ *  | Annie M. G. | Schmidt   | 20  |
+ *  </pre>
+ * can be transformed to class:
+ * <pre>
+ *
+ *          public class User {
+ *
+ *              String firstName;
+ *              String lastName;
+ *              int age;
+ *
+ *              public User() {
+ *
+ *              }
+ *          }
+ *
+ *  </pre>
+ */
+public class SimpleDefaultDataTableEntryTransformer implements DefaultDataTableEntryTransformer {
+
+	public interface InstanceCreator {
+
+		<T> T createInstance(Class<T> type) throws Throwable;
+	}
+
+	public static class DefaultInstanceCreator implements InstanceCreator {
+
+		@Override
+		public <T> T createInstance(Class<T> type) throws Throwable {
+			return type.newInstance();
+		}
+	}
+
+	private CamelCaseStringConverter fieldNameConverter = new CamelCaseStringConverter();
+	private InstanceCreator instanceCreator;
+
+	public SimpleDefaultDataTableEntryTransformer(InstanceCreator instanceCreator) {
+		this.instanceCreator = instanceCreator;
+	}
+
+	@Override
+	public <T> T transform(Map<String, String> value, Class<T> type, TableCellByTypeTransformer cellTransformer) throws Throwable {
```

Comment:
Why not use the ObjectMapper from Jackson instead? It has plugins that allow you to change the name-to-field conversion.

Reply:
I'm not Jackson expert but I couldn't make it work with class without any annotations. Eventually I'd like to use it on Android and use `Unsafe` or reflection to instantiate object using constructor with default null values. `SimpleDefaultDataTableEntryTransformer` does not need to be part of datatable project if this is a problem, I just wanted to provide simple implementation for people, but can remove it from here

Reply:
No worries. Ill try and come up with a nice way to structure this.